group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/OptimizedUniformAssignmentBuilderTest.java

```diff
@@ -0,0 +1,1090 @@
+            Map<Uuid, Set<Integer>> computedAssignmentForMember = computedGroupAssignment
+                .members().get(member).targetPartitions();
+            int sum = computedAssignmentForMember.values().stream().mapToInt(Set::size).sum();
+            totalAssignmentSizesOfAllMembers.add(sum);
+        });
+
+        for (int i = 0; i < numMembers; i++) {
+            String memberId = membersList.get(i);
+            Map<Uuid, Set<Integer>> computedAssignmentForMember =
+                computedGroupAssignment.members().get(memberId).targetPartitions();
+            // Each member is subscribed to topics of all the partitions assigned to it.
+            computedAssignmentForMember.keySet().forEach(topicId -> {
+                // Check if the topic exists in the subscription.
+                assertTrue(members.get(memberId).subscribedTopicIds().contains(topicId),
+                        "Error: Partitions for topic " + topicId + " are assigned to member " + memberId +
+                                " but it is not part of the members subscription ");
+            });
+
+            for (int j = i + 1; j < numMembers; j++) {
+                String otherMemberId = membersList.get(j);
+                Map<Uuid, Set<Integer>> computedAssignmentForOtherMember = computedGroupAssignment
+                    .members().get(otherMemberId).targetPartitions();
+                // Each partition should be assigned to at most one member
+                computedAssignmentForMember.keySet().forEach(topicId -> {
+                    Set<Integer> intersection = new HashSet<>();
+                    if (computedAssignmentForOtherMember.containsKey(topicId)) {
+                        intersection = new HashSet<>(computedAssignmentForMember.get(topicId));
+                        intersection.retainAll(computedAssignmentForOtherMember.get(topicId));
+                    }
+                    assertTrue(
+                        intersection.isEmpty(),
+                        "Error : Member 1 " + memberId + " and Member 2 " + otherMemberId +
+                            "have common partitions assigned to them " + computedAssignmentForOtherMember.get(topicId)
+                    );
+                });
+
+                // Difference in the sizes of any two partitions should be 1 at max
+                int size1 = totalAssignmentSizesOfAllMembers.get(i);
+                int size2 = totalAssignmentSizesOfAllMembers.get(j);
+                assertTrue(
+                    Math.abs(size1 - size2) <= 1,
+                    "Size of one assignment is greater than the other assignment by more than one partition "
+                        + size1 + " " + size2 + "abs = " + Math.abs(size1 - size2)
+                );
+            }
+        }
+    }
+
+    /**
+     * Verifies that the expected assignment is equal to the computed assignment for every member in the group.
+     */
+    private void assertAssignment(
```

Comment:
nit: We have the same method in RangeAssignorTest. Could we somehow share it for the two suites? We could perhaps introduce an AssignorTestUtil class in this package and add it there.

Reply:
yessir done :)